cime_config/machines/cmake_macros/gnu.cmake

```diff
@@ -40,6 +40,9 @@ endif()
 if (NOT DEBUG)
   string(APPEND FFLAGS " -O")
 endif()
+if (CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+   string(APPEND FFLAGS " -fallow-argument-mismatch -fallow-invalid-boz")
```

Comment:
We also found that we needed the "-fallow-argument-mismatch" flag for building SCORPIO with gnu 11

Reply:
I suspect you need it for GNU v10 as well